src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -838,17 +840,34 @@ def get_prompt_ids(self, text: str, return_tensors="np"):
         batch_encoding.convert_to_tensors(tensor_type=return_tensors)
         return batch_encoding["input_ids"]
 
-    @staticmethod
-    def _strip_prompt(token_ids: List[int], prompt_token_id: int, decoder_start_token_id: int):
-        has_prompt = isinstance(token_ids, list) and token_ids and token_ids[0] == prompt_token_id
+    def _strip_prompt(
+        self,
+        token_ids: Union[List[int], np.ndarray, torch.Tensor, tf.Tensor],
+        prompt_token_id: int,
+        decoder_start_token_id: int,
+    ):
+        has_prompt = (
+            isinstance(token_ids, (List[int], np.ndarray, torch.Tensor, tf.Tensor))
+            and token_ids
+            and token_ids[0] == prompt_token_id
+        )
         if has_prompt:
+            if not isinstance(token_ids, list):
+                token_ids = self._convert_to_list(token_ids)
```

Comment:
Let's convert the token ids to a list **before** checking whether we have a prompt. If we convert the token ids to a list first, then we don't need to check:
```python
isinstance(token_ids, (List[int], np.ndarray, torch.Tensor, tf.Tensor))
```
=> since we know `token_ids` will be a list!

Reply:
This is still valid! We need to convert to list **before** checking whether we have a prompt